tests/components/binary_sensor/test_timed_state_infer.py

```diff
@@ -0,0 +1,70 @@
+"""The test for the timed state infer binary sensor platform."""
+import unittest
+from unittest.mock import patch
+from datetime import timedelta
+
+from homeassistant.setup import setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import get_test_home_assistant, fire_time_changed
+
+
+class TestTimedStateInferBinarySensor(unittest.TestCase):
```

Comment:
Don't write tests that use TestCase. Instead write independent functions and have them take `hass` as an argument. That will inject a test instance. Like this: https://github.com/home-assistant/home-assistant/blob/dev/tests/components/nest/test_config_flow.py#L12